lib/src/block_parser.dart

```diff
@@ -107,11 +111,28 @@ class BlockParser {
 
   List<Node> parseLines() {
     final blocks = <Node>[];
+
+    // If the `_pos` does not change before and after `parse()`, never try to
+    // parse the line at `_pos` with the same syntax again.
+    // For example the `TableSyntax` might not advance the `_pos` in `parse`
+    // method, beause of the header row does not match the delimiter row in the
+    // number of cells, which makes a table like structure not be recognized.
+    BlockSyntax? neverMatch;
+
     while (!isDone) {
       for (final syntax in blockSyntaxes) {
+        if (neverMatch != null && neverMatch == syntax) {
```

Comment:
I think we don't need to check `neverMatch != null` separate from `neverMatch == syntax` since `syntax` is never null.

Reply:
Thanks. Fixed!